api/routers/chat.py

```diff
@@ -1,82 +1,118 @@
-from fastapi import APIRouter
-from fastapi.responses import StreamingResponse
+from fastapi import APIRouter, HTTPException, BackgroundTasks
```

Comment:
Only nitpick is you can probably get rid of the background tasks import here, but looks good now